OpenSim/Common/Test/testTRCFileAdapter.cpp

```diff
@@ -45,50 +45,76 @@ void compareFiles(const std::string& filenameA,
     int lcnt = 0;
 
     while (fileA && fileB) {
-        auto tokensA = sfa.getNextLine(fileA, delims);
-        auto tokensB = sfa.getNextLine(fileB, delims);
+        auto tokensA = OpenSim::FileAdapter::getNextLine(fileA, delims);
+        auto tokensB = OpenSim::FileAdapter::getNextLine(fileB, delims);
 
         ++lcnt;
 
+        if (tokensA.size() != tokensB.size() && lcnt < 6) {
+            // original could have any number of tabs and spaces
+            // that are no longer allowed. So ignore them.
+            OpenSim::IO::eraseEmptyElements(tokensA);
+            OpenSim::IO::eraseEmptyElements(tokensB);
+        }
+
+        if (tokensA.size() != tokensB.size()) {
+            //if a blank row, skip it
+            if (tokensA.empty() || tokensA.at(0).empty()) {
+                tokensA = OpenSim::FileAdapter::getNextLine(fileA, delims);
+                ++lcnt;
+            }
+            if (tokensB.empty() || tokensB.at(0).empty()) {
+                tokensB = OpenSim::FileAdapter::getNextLine(fileB, delims);
+                ++lcnt;
+            }
+        }
+
         std::string msg{ "Number of elements at line " +
-            std::to_string(lcnt) + "did not match." };
+            std::to_string(lcnt) + " did not match." };
 
         OPENSIM_THROW_IF(tokensA.size() != tokensB.size(),
             OpenSim::Exception, msg);
 
         for (size_t i = 0; i < tokensA.size(); ++i) {
-            if (tokensA[i] != tokensB[i]) {
-                std::string tokenA{ tokensA[i] };
-                std::string tokenB{ tokensB[i] };
-                OpenSim::IO::TrimWhitespace(tokenA);
-                OpenSim::IO::TrimWhitespace(tokenB);
-                // We interpreted blank as NaN now revert
-                // to compare to the original file with blanks
-                if (tokenB == "nan") {
-                    tokenB = "";
-                }
-                if (tokenA != tokenB) {
-                    double d_tokenA{};
-                    double d_tokenB{};
-                    try {
-                        d_tokenA = std::stod(tokenA);
-                        d_tokenB = std::stod(tokenB);
-                    }
-                    catch (std::invalid_argument&) {
-                        testFailed(filenameA, tokenA, tokenB);
-                    }
-                    if ((d_tokenA != d_tokenB) &&
-                        !(std::isnan(d_tokenA) &&
-                          std::isnan(d_tokenB))) {
-                        testFailed(filenameA, tokenA, tokenB);
-                    }
-                }
+            if (tokensA[i] == tokensB[i]) {
+                continue;
+            }
+            std::string tokenA{ tokensA[i] };
+            std::string tokenB{ tokensB[i] };
+            OpenSim::IO::TrimWhitespace(tokenA);
+            OpenSim::IO::TrimWhitespace(tokenB);
+            tokenA = OpenSim::IO::Lowercase(tokenA);
+            tokenB = OpenSim::IO::Lowercase(tokenB);
```

Comment:
Yes this was mainly intended for the NaN case.  I didn't think preserving case would be an issue since the only code that can changes case is for NaNs.  I can make it more explicit for NaNs.